Cubical/ZCohomology/Properties.agda

```diff
@@ -199,6 +213,44 @@ leftInv (Iso-coHom-coHomRed {A = A , a} n) =
                                   ; (j = i1) → 0ₖ (2 + n)})
                         (0ₖ (2 + n))
 
++∙≡+ : (n : ℕ) {A : Pointed ℓ} (x y : coHomRed (suc n) A)
+     → Iso.fun (Iso-coHom-coHomRed n) (x +ₕ∙ y)
+      ≡ Iso.fun (Iso-coHom-coHomRed n) x +ₕ Iso.fun (Iso-coHom-coHomRed n) y
+
++∙≡+ zero = sElim2 (λ _ _ → isOfHLevelPath 2 § _ _) λ _ _ → refl
++∙≡+ (suc n) = sElim2 (λ _ _ → isOfHLevelPath 2 § _ _) λ _ _ → refl
+
+private
+  homhelp : ∀ {ℓ} (n : ℕ) (A : Pointed ℓ) (x y : coHom (suc n) (typ A))
+          → Iso.inv (Iso-coHom-coHomRed {A = A} n) (x +ₕ y)
+           ≡ Iso.inv (Iso-coHom-coHomRed n) x +ₕ∙ Iso.inv (Iso-coHom-coHomRed n) y
+  homhelp n A = morphLemmas.isMorphInv _+ₕ∙_ _+ₕ_
+                (Iso.fun (Iso-coHom-coHomRed n)) (+∙≡+ n) _
+                (Iso.rightInv (Iso-coHom-coHomRed n)) (Iso.leftInv (Iso-coHom-coHomRed n))
+
+coHomGr≅coHomRedGr : ∀ {ℓ} (n : ℕ) (A : Pointed ℓ)
+                  → GroupEquiv (coHomRedGrDir (suc n) A) (coHomGr (suc n) (typ A))
+GroupEquiv.eq (coHomGr≅coHomRedGr n A) = isoToEquiv (Iso-coHom-coHomRed n)
+GroupEquiv.isHom (coHomGr≅coHomRedGr n A) = +∙≡+ n
+
+private
+  coHomGroup≡coHomRedGroup' : ∀ {ℓ} (n : ℕ) (A : Pointed ℓ)
+                          → _ ≡ coHomGroup (suc n) (typ A)
+  coHomGroup≡coHomRedGroup' n A =
+    sym (InducedAbGroupPath (coHomGroup (suc n) (typ A))
```

Comment:
@cangiuli Here's the nice use of the SIP for AbGroup that I mentioned the other day. @aljungstrom had to do some hacks to make Agda accept the definition quickly, but apart from this it's a nice application. I should try to figure out why it's so slow to use without hacks

Reply:
@mzeuner and @ecavallo might also enjoy this

Reply:
Ignoring the hacks we're showing that the reduced and unreduced Z-cohomology groups are equal as abelian groups. To make our lives easier we provide a direct definition of addition on reduced cohomology and it's then only the axioms that get transported over from the unreduced groups

Reply:
Oh wups, I just realised I don't need the hack anymore. Defining AbGroup->Group by copatterns was enough for Agda to be happy. I'll make a new wee PR

Reply:
Wow, awesome!

Reply:
Ah sorry, I think I was to trigger-happy. Agda is doing something very odd right now... I'll open a draft PR and show you